appdev/guides/pushnotifications.rst

```diff
@@ -0,0 +1,267 @@
+Push notifications
+==================
+
+Let's assume that you have an app created with Clickable and published on the OpenStore. But now you want to be able to send Push Notifications to your users. First of all, you need to understand how this is working:
+
+.. image:: ../../_static/images/appdev/guides/pushnotifications/Diagram.png
+
+1. Every app which uses push notifications has got a unique token. This token identifies the user, the device and the app itself. The token is generated by the UBports Push Service.
+2. You will need the token to send a push notification. So the app sends its token to the app developer's server.
+3. With the token you can send a HTTP request to the UBports Push Server which will forward the notification the user's device.
+
+Let's practice this step-by-step.
+
+.. note::
+    In the following example we will not implement a server. Also the communication between your app and your server is up to you. Please inform the user about the communication with your server by providing a privacy policy!
+
+Make the app ready for push notifications
+-----------------------------------------
+
+Implementing the PushClient
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+First we need to add the policy group "push-notification-client". Your apparmor file could look like this:
+
+.. code-block:: js
+    :emphasize-lines: 4
+
+	{
+	    "policy_groups": [
+		"networking",
+		"push-notification-client"
+	    ],
+	    "policy_version": 16.04
+	}
+
+In the next step we need to modify the Qml parts. We need to add a pushclient component:
+
+.. code-block:: js
+
+	//...
+
+	import Ubuntu.PushNotifications 0.1
+
+	//...
+
+	PushClient {
+		id: pushClient
+		appId: "myapp.yourname_hookname"
+		onTokenChanged: console.log("👍", pushClient.token)
+	}
+
+You need to set the correct appId! If the app name in your manifest file is myapp.yourname and the name of the main hook (the one which handles the .desktop file) is hookname, then the appId is:  myapp.yourname_hookname.
+When we now start the app, it will get a token and print this token in the logs. With clickable logs we will be able to copy this token out of the terminal. But the app is not yet ready to receive a push notification. For this we need something called a pushhelper!
+
+Implementing the pushhelper
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The pushhelper is a part of the app which will receive all push notifications and process them before sending them to the system notification center. It will receive a json-file and must output another json-file in the correct format. The pushhelper is seperated from the app. So we need a new hook in the manifest. It could look like this:
+
+.. code-block:: js
+    :emphasize-lines: 12,13
+
+	{
+	    //...
+
+	    "title": "myapp",
+	    "hooks": {
+		"myapp": {
+		    "apparmor": "myapp.apparmor",
+		    "desktop":  "myapp.desktop"
+		},
+		"push": {
+		    "apparmor": "push-apparmor.json",
+		    "push-helper": "push.json"
+		}
```

Comment:
Should lines 10-13 be highlighted instead of 12 and 13 here?

Also, is it strictly required that the myapp.apparmor has the push-notification-client if the push helper has it?